openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java

```diff
@@ -317,11 +304,12 @@ private void addResourceParameters(FunctionDefinitionNode resource, OperationAda
      */
     private void createParametersModel(FunctionDefinitionNode resource, Operation operation) {
         List<Parameter> parameters = new LinkedList<>();
-            //get path parameters
+        String in;
```

Comment:
Any specific reason to keep outside the 2 for loops ? since they are not used outside the for loops. I think can get ride of the `in` variable completely and pass `path` and `query` values directly to `buildParamater`. What do you think ?

Reply:
@lnash94 Also consider refactoring these parameter type usages using enums/constants, as I'm seeing hard-coded strings usages in many places.